ports/rp2/machine_rtc.c

```diff
@@ -100,7 +101,10 @@ STATIC mp_obj_t machine_rtc_datetime(mp_uint_t n_args, const mp_obj_t *args) {
             .sec = mp_obj_get_int(items[6]),
         };
 
-        rtc_set_datetime(&t);
+        ret = rtc_set_datetime(&t);
```

Comment:
The usual style in MicroPython (C coding) is to declare variables (eg `ret` here) as late as possible.  You could also just write it as `if (!rtc_set_datetime(&t)) { ...`.

Reply:
@dpgeorge good point. I have uploaded the fixed version.